### ISSUE TITLE
 Maximum call stack size exceeded when packing big array

### ISSUE THREAD

**Issue by wclr** [xc-jp/purescript-protobuf#38]
```proto
message BigArr {
  repeated double arr = 1;  
}
```
with code
```purs
      execPutM $ S.putBigArr $ S.mkBigArr
        {
         arr: Array.replicate 10000 1.0
        }
    
```
causes stack overflow:
```
RangeError: Maximum call stack size exceeded
    at file:///D:/Code/proj/output/Control.Monad.Writer.Trans/index.js:103:45
    at file:///D:/Code/proj/output/Control.Monad/index.js:70:33
    at file:///D:/Code/proj/output/Effect/foreign.js:10:14
    at file:///D:/Code/proj/output/Effect/foreign.js:10:20
    at file:///D:/Code/proj/output/Effect/foreign.js:10:20
    at file:///D:/Code/proj/output/Effect/foreign.js:10:16
    at file:///D:/Code/proj/output/Effect/foreign.js:10:20
    at file:///D:/Code/proj/output/Effect/foreign.js:10:20
    at file:///D:/Code/proj/output/Effect/foreign.js:10:16
    at file:///D:/Code/proj/output/Effect/foreign.js:10:16
```

**Comment by jamesdbrock** [comment]
Oh yeah that's a bug, thanks for the report.

https://github.com/xc-jp/purescript-protobuf/blob/dbe4d5039b8bd73b72936d22aba483ab642dfc5a/src/Protobuf/Internal/Runtime.purs#L226

should not use `traverse_`, and instead should use the stack-safe `foldRecM_`.

**Comment by jamesdbrock** [comment]
I will try to publish a new release today.

**Comment by wclr** [comment]
It would be great.

**Comment by jamesdbrock** [comment]
I published [v4.4.0](https://pursuit.purescript.org/packages/purescript-protobuf/4.4.0) which should fix that bug.
